packages/nuxt/src/app/config.ts

```diff
@@ -0,0 +1,53 @@
+import type { AppConfig, RuntimeConfig, AppConfigInput } from '@nuxt/schema'
+import { reactive } from 'vue'
+import { useNuxtApp } from './nuxt'
+// @ts-ignore
+import __appConfig from '#build/app.config.mjs'
+
+// Workaround for vite HMR with virtual modules
+export const _appConfig = __appConfig as AppConfig
+
+export function useRuntimeConfig (): RuntimeConfig {
+  return useNuxtApp().$config
+}
+
+export function useAppConfig (): AppConfig {
+  const nuxtApp = useNuxtApp()
+  if (!nuxtApp._appConfig) {
+    nuxtApp._appConfig = reactive(_appConfig) as AppConfig
+  }
+  return nuxtApp._appConfig
+}
+
+// HMR Support
+if (process.dev) {
+  function applyHMR (newConfig: AppConfigInput) {
```

Comment:
This should be the resolved config:

```suggestion
  function applyHMR (newConfig: AppConfig) {
```

Reply:
... not that makes a huge difference as it's internal!